Quote escape bug fix

kahnvex opened the pull request:
Credit to @beck for telling me lodash templates were not the right choice. I've lifted code from gulp-static-i18n that needs to be modularized and made installable. This should fix most of the possible bugs we'd end up having.

kahnvex committed:
This fixes a quote escaping bug

Credit to @beck for telling me lodash templates were not the right choice. I've lifted code from gulp-static-i18n that needs to be modularized and made installable. This should fix most of the possible bugs we'd end up having.

snitch commented on index.js:
```diff
@@ -14,19 +14,31 @@ var extension = function(module, filename) {
+  return str
     .replace(/\n/g, '\\n')
     .replace(/"/g, '\\"');
+}
```
Missing semicolon.

kahnvex committed:
Alphabetize var declarations

kahnvex committed:
Add missing semicolon

snitch commented:
:sparkles: No lint errors found. :sparkles:

euwest commented on index.js:
```diff
@@ -14,19 +14,31 @@ var extension = function(module, filename) {
+  var re = opts.interpolate || /\{\{trans\s*(?:"([^"]+)"|\'([^\']+)\')\s*\}\}/g;
+  var chunkString, match, msgid, needle, translated;
+
+  translated = template;
```
I don't see the _point_ of this(badum-tss)

euwest replied:
actually i see the point, it's line 35 that doesn't make sense. i think it should be `match = re.exec(translated);`

euwest replied:
nevermind, i just saw `Regex.prototype.exec()`'s successive match feature. that's dope

beck commented:
I think this is better than using lodash (and it's probably hellafast) but now we're stepping into regex hell.  I think the only thing worse then regex'ing sourcecode is regex'ing for regexes.

Doing a quick browse of handlebars the handlebars source, I was hoping to find something we can pluck.  Then I saw their [quotedString()](https://github.com/wycats/handlebars.js/blob/d4070c36675bfecee290f20bd2d9c23a50e9e00b/lib/handlebars/compiler/code-gen.js#L101-L109) ... and I ran away.  Writing good parsers is hard.

I still recommend using handlebars and suffer the slower builds, but that is unfair of me to say since I'm not responsible for implementing this.  It's a call for @kahnjw but please consider the time of debugging regex vs slower builds.

kahnjw commented:
>I still recommend using handlebars and suffer the slower builds, but that is unfair of me to say since I'm not responsible for implementing this. It's a call for @kahnjw but please consider the time of debugging regex vs slower builds.

It's not only that it is slower, but it is also that doing this would not allow consumers to use the Handlebars runtime in their application. We only want to replace translation tags, not every `{{ ... }}`.

euwest commented:
`quotedString` looks like what we want for our escape function, actually...

beck commented:
> We only want to replace translation tags, not every {{ ... }}.

Ahh, since this package is converting the template to a module, I assumed this package is responsible for the hbs > js compilation and any additional handlebar manipulation was nixed. My mistake.

kahnvex committed:
Initialize chunkString as emtpy string

snitch commented:
:sparkles: No lint errors found. :sparkles:

kahnvex committed:
Return a valid catalog when locale is default

When the locale is default, 'en', we need to return a valid catalog. This makes that happen. Also, we'll want to make the default locale configurable, I'll follow up on this soon.

Issue: https://github.com/yola/browserify-i18n/issues/5

snitch commented on index.js:
```diff
@@ -41,8 +54,10 @@ var filterInterpolator = function(file, catalog, opts) {
-    var fp = path.join(localeDir, locale, 'LC_MESSAGES', 'messages.po');
+  var poParser = function(localeDir, index, localeDirs, en, filename) {
+    locale = en || locale;
+    var filename = filename || 'messages.po';
```
'filename' is already defined.

kahnvex committed:
Don't redefine filename

snitch commented:
:sparkles: No lint errors found. :sparkles:

kahnjw commented:
Closing in favor of #6
